ts/containers/connect_wallet_dialog.ts

```diff
@@ -28,6 +32,9 @@ const mapDispatchToProps = (dispatch: Dispatch<State>): ConnectedDispatch => {
         onDismiss: (): void => {
             dispatcher.updateIsConnectWalletDialogOpen(false);
         },
+        onConnectWallet: (provider: Providers): void => {
+            dispatcher.connectWallet(provider);
```

Comment:
The main advantages with sagas is that you get to model your async side effects into isolated worker tasks. They react to actions and can trigger state changes as well as new actions. The components just have to worry about triggering an action and the saga will take care of handling the side effect(s) of that action.

We could just make the container method async let that handle the async processing and dispatching the appropriate response and if that is an established pattern in other places it may be preferable.

The slight downside of this would be:
- the component now needs to be aware of the asynchronous nature of the dispatcher function and need to handle that appropriately
- it may get trickier to combine multiple side effects. In Sagas one saga can trigger multiple other sagas and so on without the component that triggered the action having to worry about it
- Sagas offer a simple way to debounce long running actions. `takeLatest` will ignore additional action triggers while the saga responding to it is already executing. This may need to be handled manually with async dispatchers.

I reached for Sagas because that is how I have previously handled async side effects in a Redux environment. I checked for existing patterns for handling this in the codebase (but only the website) and didn't find any so I did not know there was a precedence available.

I don't really have any strong opinions about how to handle this, if you think it worked well the way you've handled it in instant I'd be happy to follow that pattern 👍 Like I said previously, it's been a while since I worked with Redux, the status quo may have changed since 😄

Reply:
I'm comfortable leaving this up to you, and you'll be doing most of the work with it in the short-medium term.

To your points (respectively):
- I'm not sure I'm understanding why the component needs to be aware of the async nature of the dispatch. Ideally the component would not know if anything is being dispatched.
- "In Sagas one saga can trigger multiple other sagas and so on without the component that triggered the action having to worry about it" -- I think this is the main selling point. In instant we don't run into too much async complexity. Most actions are of the form "I'm doing some async things" ->"Do the async things" -> "Done with my list of async things". I think for this, async/await is enough. You can share logic with an async function and it is easy to understand. Do you think we will be composing async actions a lot? I think that is the real question, if not then async/await is enough IMHO.
- Yes this makes sense, but if we don't have async actions triggering async actions, then "debounce" is enough for straight up handlers.

Reply:
Let's try with async dispatcher functions. Nice to have consistency between the projects and it's easy enough move to Sagas later on if needed. I'm all for minimising external dependency reliance 😄